### PR TITLE
BMS-370: Add ability to get profile based on authentication headers

### PR DESCRIPTION
**Pull request by vartanbeno**
[BMS-370](https://jira.touchtunes.com/browse/BMS-370)

### CODE REVIEW

File: test/lib/provision.js [hunk 1]
```diff
@@ -550,26 +550,26 @@ describe('provision', () => {
...
-		it('should require a profileAlias (promise)', (done) => {
-			provision.getProfile()
-				.then(() => {
-					return done(new Error('should require profileAlias'));
-				})
-				.catch((err) => {
-					should.exist(err);
-					should.exist(err.message);
-					err.message.should.contain('profileAlias is required');
+		it('should properly retrieve profile even without profileAlias', (done) => {
```
Comment (badaldesai) [inline test/lib/provision.js]: Not sure, what is being tested here. If it is same as next test, this one might be removed.

**Review by badaldesai — CHANGES_REQUESTED** [review]

File: test/lib/provision.js (hunk 1 above)
Comment (vartanbeno) [reply test/lib/provision.js]: Oh, there's a test for the promise-based result and the callback-based one.

Comment (Mig-OG) [reply test/lib/provision.js]: From mocha documentation, it is discourage to use arrow function. https://mochajs.org/#arrow-functions. I know there is a lot of test written that way, but let's not add some. Please change from arrow function to old style lambda: `function() {}`

File: test/lib/provision.js [hunk 2]
```diff
@@ -550,26 +550,26 @@ describe('provision', () => {
...
 
-					return done();
-				});
+			provision.getProfile()
+				.then(() => done())
+				.catch((err) => done(err));
 		});
 
-		it('should require profileAlias (callback)', (done) => {
+		it('should properly retrieve profile even without profileAlias (callback)', (done) => {
```
Comment (Mig-OG) [inline test/lib/provision.js]: arrow function to old style lambda

**Review by Mig-OG — CHANGES_REQUESTED** [review]

File: test/lib/provision.js (hunk 2 above)
Comment (vartanbeno) [reply test/lib/provision.js]: Done

**Review by badaldesai — APPROVED** [review]

**Review by Mig-OG — APPROVED** [review]

**Review by akobyshchatt — APPROVED** [review]

**Pull request merged**